openff/bespokefit/schema/tasks.py

```diff
@@ -1,28 +1,21 @@
 import abc
-from typing import Tuple
+from typing import Optional, Tuple
 
+from openff.qcsubmit.procedures import GeometricProcedure
 from pydantic import Field, conint
 from qcelemental.models.common_models import Model
 from typing_extensions import Literal
 
 from openff.bespokefit.utilities.pydantic import BaseModel
 
 
-class OptimizationSpec(BaseModel):
-
-    procedure: Literal["geometric"] = "geometric"
-
-    max_iterations: conint(gt=0) = Field(
-        300,
-        description="The maximum number of iterations to perform before raising a "
-        "convergence failure exception.",
-    )
-
-
 class QCGenerationTask(BaseModel, abc.ABC):
 
     type: Literal["base-task"]
 
+    program: str = Field(..., description="The program to use to evaluate the model.")
+    model: Model = Field(..., description=str(Model.__doc__))
+
```

Comment:
Did you want to replace these with the QCSpec model at some point?

Reply:
Yes I think the validation in the QCSpec model should help avoid a lot of errors Ill swap these fields out for a `qc_spec` field.

Reply:
This is done in [df5a6bf](https://github.com/openforcefield/bespoke-fit/pull/77/commits/df5a6bfdab02fc000479ab3c405e944b1e16744a) but this means that the qc cache now depends on arbitrary fields like the spec name and spec description, maybe the QCSpec object should be split in qcsubmit to offer a version without these fields but which still offers the validation?